docs/developer-guide/plugins-scenarios.md

```diff
@@ -98,25 +98,25 @@ public class AsyncRequestInterceptor implements Plugin {
 }
 ```
 
-Step 1. We call the `asyncUrlReplacement`, which returns a `StyxObservable<Url>`.
+Step 1. We call the `asyncUrlReplacement`, which returns an `Eventual<Url>`.
 The `asyncUrlReplacement` wraps a call to the remote service and converts 
-the outcome into a `StyxObservable`, which is the basis for our response observable.
+the outcome into a `Eventual`.
 
 Step 2. A call to `pathReplacementService` makes a non-blocking call to the remote key/value store.
 Well, at least we pretend to call the key value store, but in this example we'll just return a 
 completed future of a constant value.   
 
-Step 3. `CompletableFuture` is converted to `StyxObservable`, so that other asynchronous
+Step 3. `CompletableFuture` is converted to an `Eventual`, so that other asynchronous
 operations like `chain.proceed` can be bound to it later on.
 
 Step 4. The eventual outcome from the `asyncUrlReplacement` yields a new, modified URL instance.
 We'll transform the `request` by substituting the URL path with the new one. 
 This is a quick synchronous operation so we'll do it in a `map` operator. 
 
-Step 5. Finally, we will bind the outcome from `chain.proceed` into the response observable.
-Remember that `chain.proceed` returns an `Observable<HttpResponse>` it is therefore 
-interface compatible and can be `flatMap`'d to the response observable. The resulting
-response observable chain is returned from the `intercept`.
+Step 5. Finally, we will bind the outcome from `chain.proceed` into the response eventual.
```

Comment:
I would suggest always using "`Eventual`" to refer to the class, rather than "eventual", as this is a particular concept for Styx, rather than the normal use of the word (which is just an adjective).

Reply:
ok